include/seqan3/io/sam_file/detail/format_sam_base.hpp

```diff
@@ -433,182 +432,270 @@ inline void format_sam_base::read_header(stream_view_type && stream_view,
         }
     };
 
-    auto skip_tag_name_and_colon = [&it] ()
+    auto skip_until_predicate = [&it] (auto const & predicate)
     {
-        while (!is_char<':'>(*it))
+        while (!predicate(*it))
             ++it;
-        ++it;
     };
 
     auto parse_tag_value = [&] (auto & value) // helper function to parse the next tag value
     {
-        skip_tag_name_and_colon();
+        skip_until_predicate(is_char<':'>);
+        ++it; // skip :
         take_until_predicate(is_char<'\t'> || is_char<'\n'>);
         read_field(string_buffer, value);
     };
 
-    auto parse_until_newline = [&] (auto & value)
+    // Some tags are not parsed individually. Instead, these are simply copied into a std::string.
+    // Multiple tags must be separated by a `\t`, hence we prepend a tab to the string, except the first time.
+    // Alternatively, we could always append a `\t`, but this would have the side effect that we might need to trim a
+    // trailing tab after parsing all tags via `pop_back()`.
+    // Unfortunately, we do not know when we are parsing the last tag (and in this case just not append a tab),
+    // because even though we can check if the line ends in a `\n`, it is not guaranteed that the last tag of the
+    // line is passed to this lambda. For example, the line might end with a tag that is properly parsed, such as `ID`.
+    auto parse_until_tab_or_newline = [&] (std::string & value, std::array<char, 2> raw_tag)
```

Comment:
```suggestion
    auto parse_unknown_tag_value = [&] (std::string & value, std::array<char, 2> raw_tag)
```

?

Reply:
Maybe even `append_tag_to_string` since it does not necessarily need to be unknown, we just use this lambda in 2 places where we "dump" all tags that get no special treatment into a string

Reply:
okay, then: parse_and_append_unknown_tag_to_string?

Reply:
for me okay

Reply:
Sure, but maybe we should remove the `unknown`, i.e. `parse_and_append_tag_to_string`; `unhandled` would probably be more correct...
for the really unknown tags we emit a warning, and skip them.

Reply:
then `parse_and_append_unhandled_tag_to_string`